### PR TITLE
[5.1] Aliasing Filesystem classes for installation, updating script.php

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #42971 .

### Summary of Changes
With the refactoring of system calls to the framework filesystem package, I introduced a new error when updating since at the time when script.php is run, the framework class ist not there. This PR changes the namespace of the classes in the finalisation.php to reflect first the framework filesystem package and then to alias those classes for the CMS filesystem package. It also fixes the code in script.php to use the framework code consistently. That should hopefully fix the issue and be compatible pretty much regardless from where to where we upgrade.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@Hackwar Does PR #42976 still make sense together with this one here? I assume yes, but maybe you know more.

File: administrator/components/com_joomlaupdate/finalisation.php [hunk 1]
```diff
@@ -236,6 +236,14 @@ public static function delete(string $folderName): bool
             }
         }
     }
+
+    if (!class_exists('\Joomla\CMS\Filesystem\File')) {
+        \JLoader::registerAlias('\\Joomla\\CMS\\Filesystem\\File', '\\Joomla\\Filesystem\\File');
```
Comment (joomdonation) [inline administrator/components/com_joomlaupdate/finalisation.php]: The code on this file is running as a standalone PHP script, it is not running inside Joomla environment, so unless I'm wrong, we could not use `JLoader::registerAlias` here.

Comment (joomdonation) [reply administrator/components/com_joomlaupdate/finalisation.php]: With the change you made to com_admin/script.php in this PR, we do not use `\Joomla\CMS\Filesystem\Folder` and `\Joomla\CMS\Filesystem\Folder` in `JoomlaInstallerScript` class anymore,  I would say you can remove the code which registers the classes alias here. Register alias is not needed (and it also does not work at all as I mentioned in above comment)

**Comment by joomdonation** [comment]
Could you please also delete the two elements here https://github.com/joomla/joomla-cms/blob/5.0-dev/administrator/components/com_admin/script.php#L575-L576 as I did in in my original PR https://github.com/joomla/joomla-cms/pull/42976 ? Thanks !

File: administrator/components/com_joomlaupdate/finalisation.php (hunk 1 above)
Comment (Hackwar) [reply administrator/components/com_joomlaupdate/finalisation.php]: You are right about the Jloader class. I'm going to fix that. Regarding the alias I would rather be safe than sorry. I fear that there is some scenario where the script.php was not updated and then this fails... Yes. I know it is not very likely, but for now lets keep it that way. Maybe we could even go one step further and simply use unlink() etc. to do these things, removing the need of these classes entirely?

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 5f35b5d57b22197e96135feac43908d4036d1b01<br /><br />Tested update from both Joomla 4 and Joomla 5 to the update package provided by this PR, it worked well, no error anymore<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42978">issues.joomla.org/tracker/joomla-cms/42978</a>.</sub>

**Comment by richard67** [comment]
> I have tested this item ✅ successfully on [5f35b5d](https://github.com/joomla/joomla-cms/commit/5f35b5d57b22197e96135feac43908d4036d1b01)Tested update from both Joomla 4 and Joomla 5 to the update package provided by this PR, it worked well, no error anymore

@joomdonation Have you used Windows or a Linux as server? Would be good if we get another tester using what you haven't used.

**Comment by richard67** [comment]
@brianteeman Does your thumbs down reaction on the description of this PR still apply, or is it obsolete after the recent changes? If obsolete: Please remove. If not obsolete: Would you like to comment on the reasons? Thanks in advance.

**Comment by joomdonation** [comment]
@richard67 I tested using Windows. I also did a careful code review.

**Comment by richard67** [comment]
> @richard67 I tested using Windows. I also did a careful code review.

@joomdonation Thanks. I did also a careful code review. PR is ok for me. Will see if I can find time to test on Linux.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 5f35b5d57b22197e96135feac43908d4036d1b01<br /><br />I've tested both live update and upload & update from 4.4.3 and 5.0.3 to the custom update URL or update package created by drone for this PR.

I've also tested that updating from 5.1.0-beta 1 with the PR applied before the update works.

Finally I've done a code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42978">issues.joomla.org/tracker/joomla-cms/42978</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42978">issues.joomla.org/tracker/joomla-cms/42978</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**